p2p/peer.go

```diff
@@ -0,0 +1,218 @@
+/**
+*  @file
+*  @copyright defined in go-seele/LICENSE
+ */
+
+package p2p
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/seeleteam/go-seele/p2p/discovery"
+)
+
+type Peer struct {
+	fd       net.Conn
+	node     *discovery.Node
+	created  uint64
+	err      error
+	closed   chan struct{}
+	disc     chan uint
+	protoMap map[uint16]*Protocol // protoCode=>proto
+	capMap   map[string]uint16    //cap of protocol => protoCode
+
+	wMutex sync.Mutex // for conn write
+	wg     sync.WaitGroup
+}
+
+func (p *Peer) run() {
+	// add peer to protocols
+	for _, proto := range p.protoMap {
+		proto.AddPeerCh <- p
+	}
+	var (
+		writeErr = make(chan error, 1)
+		readErr  = make(chan error, 1)
+		//reason   uint // sent to the peer
+		err error
+	)
+	p.wg.Add(2)
+	go p.readLoop(readErr)
+	go p.pingLoop()
+
+	// Wait for an error or disconnect.
+loop:
+	for {
+		select {
+		case err = <-writeErr:
+			// A write finished. Allow the next write to start if
+			// there was no error.
```

Comment:
nice to refine the comments:
When a write operation finished without any error, the next write operation will continue.